### ISSUE TITLE
Run tests asynchronously

### ISSUE THREAD

**Issue by Leandros** [catchorg/Catch2#1308]
This is a big one, I know, but with increasing number of tests I found catch to become very slow. 
Since the tests in each `TEST_CASE` are (or rather should) be self-contained, it would be great if there was a knob to turn which runs all tests in parallel (up to N, where N is usually number of cores).

This would require a couple of changes I presume, since not even `REQUIRE` on it's own is thread-safe.

**Comment by Leandros** [comment]
I've seen this has been requested in the past, and is mentioned in the known limitations. I'd still like to nag you. The current solutions are not satisfying.

**Comment by horenmar** [comment]
Not happening.

To expand on it a bit, we have decided that having parallel _runner_ is out of scope for Catch, and getting multiple tests to run in parallel should instead be left for external test runners.

We do however plan to have (optionally?) thread-safe assertions at some point.

**Comment by JoeyGrajciar** [comment]
Closing based on last comment
